diff -r ec5e51fd886c src/dm/impls/mesh/section.c
--- a/src/dm/impls/mesh/section.c	Sat Mar 09 10:50:47 2013 -0600
+++ b/src/dm/impls/mesh/section.c	Tue Jan 28 17:49:49 2014 -0600
@@ -296,6 +296,8 @@
   PetscFunctionBegin;
   if (!*section) PetscFunctionReturn(0);
   PetscValidHeaderSpecific(*section, SECTIONREAL_CLASSID, 1);
+  (*section)->s = NULL;
+  (*section)->b = NULL;
   if (--((PetscObject)(*section))->refct > 0) PetscFunctionReturn(0);
   ierr = PetscHeaderDestroy(section);CHKERRQ(ierr);
   PetscFunctionReturn(0);

