diff --git a/include/petscsnes.h b/include/petscsnes.h
--- a/include/petscsnes.h
+++ b/include/petscsnes.h
@@ -607,6 +607,7 @@ PETSC_EXTERN PetscErrorCode SNESLineSear
 /* Routines for VI solver */
 PETSC_EXTERN PetscErrorCode SNESVISetVariableBounds(SNES,Vec,Vec);
 PETSC_EXTERN PetscErrorCode SNESVISetComputeVariableBounds(SNES, PetscErrorCode (*)(SNES,Vec,Vec));
+PETSC_EXTERN PetscErrorCode SNESVIGetVariableBounds(SNES,Vec*,Vec*);
 PETSC_EXTERN PetscErrorCode SNESVIGetInactiveSet(SNES,IS*);
 PETSC_EXTERN PetscErrorCode SNESVIGetActiveSetIS(SNES,Vec,Vec,IS*);
 PETSC_EXTERN PetscErrorCode SNESVIComputeInactiveSetFnorm(SNES,Vec,Vec,PetscReal*);
diff --git a/src/snes/impls/vi/vi.c b/src/snes/impls/vi/vi.c
--- a/src/snes/impls/vi/vi.c
+++ b/src/snes/impls/vi/vi.c
@@ -549,6 +549,33 @@ PetscErrorCode SNESVISetVariableBounds_V
 }
 EXTERN_C_END
 
+#undef __FUNCT__
+#define __FUNCT__ "SNESVIGetVariableBounds"
+/*@
+   SNESVIGetVariableBounds - Sets the lower and upper bounds for the solution vector. xl <= x <= xu.
+
+   Input Parameter:
+.  snes - the SNES context.
+
+   Output Paremeters: 
+.  xl   - lower bound.
+.  xu   - upper bound.
+
+   Level: advanced
+
+@*/
+PetscErrorCode SNESVIGetVariableBounds(SNES snes, Vec *xl, Vec *xu)
+{
+  PetscFunctionBegin;
+  if (snes->usersetbounds) {
+    *xl = snes->xl;
+    *xu = snes->xu;
+  } else {
+    SETERRQ(PETSC_COMM_SELF,PETSC_ERR_ARG_WRONGSTATE,"Must set SNESVI bounds");
+  }
+  PetscFunctionReturn(0);
+}
+
 #undef __FUNCT__  
 #define __FUNCT__ "SNESSetFromOptions_VI"
 PetscErrorCode SNESSetFromOptions_VI(SNES snes)
