*** petsc-3.3-p7/src/dm/impls/mesh/ftn-auto/meshf.c	2013-05-11 22:17:17.000000000 -0500
--- petsc-3.3/src/dm/impls/mesh/ftn-auto/meshf.c	2014-01-23 14:15:12.309856687 -0600
*************** extern void PetscRmPointer(void*);
*** 133,138 ****
--- 133,143 ----
  #elif !defined(PETSC_HAVE_FORTRAN_UNDERSCORE) && !defined(FORTRANDOUBLEUNDERSCORE)
  #define dmmeshassemblematrix_ dmmeshassemblematrix
  #endif
+ #ifdef PETSC_HAVE_FORTRAN_CAPS
+ #define dmmeshclone_ DMMESHCLONE
+ #elif !defined(PETSC_HAVE_FORTRAN_UNDERSCORE) && !defined(FORTRANDOUBLEUNDERSCORE)
+ #define dmmeshclone_ dmmeshclone
+ #endif
  
  
  /* Definitions of Fortran Wrapper routines */
*************** void PETSC_STDCALL  dmmeshassemblematrix
*** 230,235 ****
--- 235,244 ----
  	(DM)PetscToPointer((dm) ),
  	(SectionReal)PetscToPointer((section) ),*e,v,*mode);
  }
+ void PETSC_STDCALL  dmmeshclone_(DM dm,DM *newdm, int *__ierr ){
+ *__ierr = DMMeshClone(
+ 	(DM)PetscToPointer((dm) ),newdm);
+ }
  #if defined(__cplusplus)
  }
  #endif
